test/framework/k8s_cluster.go

```diff
@@ -515,28 +519,42 @@ func (c *K8sCluster) DeployKuma(mode core.CpMode, opt ...KumaDeploymentOption) e
 		return err
 	}
 
-	err = c.WaitApp(Config.KumaServiceName, Config.KumaNamespace, replicas)
-	if err != nil {
-		return err
-	}
-
+	appsToInstall := make([]appInstallation, 0)
+	appsToInstall = append(appsToInstall, appInstallation{Config.KumaServiceName, Config.KumaNamespace, replicas})
 	if c.opts.cni {
-		err = c.WaitApp(Config.CNIApp, Config.CNINamespace, 1)
-		if err != nil {
-			return err
-		}
+		appsToInstall = append(appsToInstall, appInstallation{Config.CNIApp, Config.CNINamespace, 1})
 	}
-
 	if c.opts.zoneIngress {
-		if err := c.WaitApp(Config.ZoneIngressApp, Config.KumaNamespace, 1); err != nil {
-			return err
-		}
+		appsToInstall = append(appsToInstall, appInstallation{Config.ZoneIngressApp, Config.KumaNamespace, 1})
 	}
-
 	if c.opts.zoneEgress {
-		if err := c.WaitApp(Config.ZoneEgressApp, Config.KumaNamespace, 1); err != nil {
-			return err
-		}
+		appsToInstall = append(appsToInstall, appInstallation{Config.ZoneEgressApp, Config.KumaNamespace, 1})
+	}
```

Comment:
Good call that should speed things up no?

Reply:
Right, should be faster by installing the apps in parallel. A drawback is it makes the log output disordered instead of sequencial.